akka-stream/src/main/scala/akka/stream/scaladsl/Source.scala

```diff
@@ -974,4 +974,27 @@ object Source {
       close: (S) => Future[Done]): Source[T, NotUsed] =
     Source.fromGraph(new UnfoldResourceSourceAsync(create, read, close))
 
+  /**
+   * Merge multiple [[Source]]s. Prefer the sources depending on the 'priority' parameters.
+   * The provided sources and priorities must have the same size and order.
+   *
+   * '''emits''' when one of the inputs has an element available, preferring inputs based on the 'priority' parameters if both have elements available
+   *
+   * '''backpressures''' when downstream backpressures
+   *
+   * '''completes''' when both upstreams complete (This behavior is changeable to completing when any upstream completes by setting `eagerComplete=true`.)
+   *
+   * '''Cancels when''' downstream cancels
+   */
+  def mergePrioritizedN[T](
+      sourcesAndPriorities: immutable.Seq[(Source[T, _], Int)],
+      eagerComplete: Boolean = false): Source[T, NotUsed] = {
```

Comment:
Default values makes keeping binary compatibility problematic, so let's skip that, if we need a default shortcut an overload with only one parameter is better.
```suggestion
      eagerComplete: Boolean): Source[T, NotUsed] = {
```